src/coreclr/jit/emit.cpp

```diff
@@ -4102,6 +4199,382 @@ void emitter::emitDispCommentForHandle(size_t handle, GenTreeFlags flag)
 #endif // DEBUG
 }
 
+//****************************************************************************
+// emitRemoveJumpToNextInst:  Checks all jumps in the jump list to see if they are
+//   unconditional jumps generated at the end of BB_ALWAYS basic blocks and whether
+//   they are now a jump to the next instruction, this can happen when the following
+//   basic block emitted no instructions
+//
+// Returns:
+//   true if any jumps have been removed, otherwise returns false
+//
+// Assumptions:
+//    the jump list must be ordered by increasing igNum+insNo
+//
+bool emitter::emitRemoveJumpToNextInst()
+{
+#ifdef TARGET_XARCH
+#ifdef DEBUG
+    if (emitComp->verbose)
+    {
+        printf("*************** In emitRemoveJumpToNextInst()\n");
```

Comment:
Use `JITDUMP` and remove `emitComp->vebose` check here and below.

Reply:
This needs to be done throughout the file. I'm not going to do it in this PR but i feel a lot of things could be tidied up quite simply in here.

Reply:
That's the policy but the style is mixed throughout the file which makes it hard to identify the correct way to do things. Old patterns are present that could easily be updated to be more consistent and thus clearer. I know that there's a strong wish to avoid large changes through the jit because of the extra work it adds to in-flight PRs but I think there is some level of cleanup that could be done at individual file level. 

In emit.cpp it would be quite simple to audit use of `emitComp->verbose` and `EMITEVERBOSE` and use 'JITDUMP' where it makes sense to do so. this would benefit readability. in emit.h instrDesc uses accessor functions but instrDescJmp doesn't. It's a simple PR to change instrDescJmp to use accessor functions and flow that through the project. As long as any changes are kept mechanical these should be simple to do and review and would ultimately make life easier for anyone working in the code.

Reply:
> In emit.cpp it would be quite simple to audit use of emitComp->verbose and EMITEVERBOSE and use 'JITDUMP' where it makes sense to do so.

Based on the top of emit.h I assume the intent of `EMITVERBOSE` is to allow easy logging of only the emitter parts of the JIT instead of enabling full jit-wide logging, so these might not be easily convertible.

For `emitComp->verbose` I agree we should convert them, but I see arguably only one or two convertible usages in the file.

> in emit.h instrDesc uses accessor functions but instrDescJmp doesn't. It's a simple PR to change instrDescJmp to use accessor functions and flow that through the project. As long as any changes are kept mechanical these should be simple to do and review and would ultimately make life easier for anyone working in the code.

I think it is up to debate whether that makes things better. Introducing hundreds of lines of accessor functions for trivial field access does not seem better to me without additional refactoring to make the emitter less "C-like". For `instrDesc` the encapsulation is more important since the structure depends on the platform so much.
I realize our guidelines say not to declare public data members, but they are guidelines for a reason and sometimes require looking at the bigger picture.